### PR TITLE
Basker solve

### PR DESCRIPTION
**Pull request by ndellingwood**
@srajama1  For review purposes. 
Changes to ShyLU Basker and Amesos2 for performance improvement during solve.

### CODE REVIEW

**Comment by ndellingwood** [comment]
@srajama1 After review (hash cf1981b to be looked at carefully regarding Amesos2) I will rebase off current Trilinos develop, squash commit messages and reissue pull request.

File: packages/amesos2/src/Amesos2_EpetraMultiVecAdapter_def.hpp [hunk 1]
```diff
@@ -222,6 +222,11 @@ void MultiVecAdapter<Epetra_MultiVector>::get1dCopy(
 		      "and number of vectors" );
 #endif
```
Comment (srajama1) [inline packages/amesos2/src/Amesos2_EpetraMultiVecAdapter_def.hpp]: Should this be done for Amesos2_TpetraMultiVecAdapter_def.hpp as well ?

File: packages/shylu/basker/src/basker_solve_rhs.hpp [hunk 2]
```diff
@@ -838,140 +553,138 @@ namespace BaskerNS
...
+      const Int iend = M.col_ptr(k+1);
+
+      y(k+brow) = x(k+bcol) / M.val(M.col_ptr(k));
+
+      const auto ykbcol = y(k+bcol);
+      //for(Int i = M.col_ptr(k)+1; i < M.col_ptr(k+1); ++i)
+      for(Int i = istart+1; i < iend; ++i)
+      {
+        //Int j = gperm[M.row_idx[i]];
+//        const Int j = gperm(M.row_idx(i)+brow); //NDE: why use gperm?
```
Comment (srajama1) [inline packages/shylu/basker/src/basker_solve_rhs.hpp]: Might be needed just when using pivoting, May be a test outside ?

Comment (srajama1) [inline packages/shylu/basker/src/basker_solve_rhs.hpp]: basker->Options.no_pivot      = BASKER_TRUE; can be used to check if gperm is needed or not

File: packages/shylu/basker/src/shylubasker_def.hpp [hunk 3]
```diff
@@ -276,296 +276,311 @@ namespace BaskerNS
...
     factor_notoken(option);
-    
-    #ifdef BASKER_KOKKOS_TIME
+
+#ifdef BASKER_KOKKOS_TIME
     stats.time_nfactor += timer.seconds();
-    #endif
+#endif
+
+    // NDE
```
Comment (srajama1) [inline packages/shylu/basker/src/shylubasker_def.hpp]: Move to so symbolic ?

**Comment by ndellingwood** [comment]
@srajama1 @hkthorn 
Made changes to incorporate most comments from pull request review (Amesos2 Tpetra single multi-vector case and moving some ShyLU Basker allocations to Symbolic will be included in a followup push). 
Rebased with current Trilinos develop branch; commit messages squashed to a single message.

**Comment by srajama1** [comment]
@ndellingwood : Thanks a lot  !

**Pull request merged**